spec/features/self_service/firms_index_spec.rb

```diff
@@ -105,6 +122,26 @@ def and_the_trading_names_section_is_showing_a_prompt_to_add_a_trading_name
     expect(firms_index_page).not_to have_trading_names
   end
 
+  def when_i_delete_the_first_trading_name
+    firms_index_page.trading_names.first.delete_button.click
+  end
+
+  def and_i_can_see_a_success_message
+    expected_message = "#{@first_trading_name_registered_name} has been successfully removed."
+    expect(firms_index_page).to have_flash_message(text: expected_message)
+  end
+
+  def and_i_can_see_the_first_trading_name
+    @trading_name_registered_name_to_delete = firms_index_page.trading_names.first.name.text
```

Comment:
`firm_index_page.trading_names.first` is coming from the view. Funnily enough the implementation earlier caused random test failures, this fixed it.